reactor-core/src/main/java/reactor/core/publisher/FluxBufferWhen.java

```diff
@@ -86,7 +86,10 @@ public void subscribe(CoreSubscriber<? super BUFFER> actual) {
 		BufferWhenOpenSubscriber<OPEN> bos = new BufferWhenOpenSubscriber<>(main);
 		if (main.subscribers.add(bos)) {
 			start.subscribe(bos);
-			source.subscribe(main);
+			return main;
```

Comment:
there should be no side effect of replacing the direct call with a return+while loop, but I would still try and come up with corner cases for these patterns where there are 2 subscriptions.

Reply:
I'm not sure we can do it with zero allocations... need to think about it

Reply:
We could defer the companion subscribe in general to onSubscribe but that is a different story.